src/components/swap/GasBreakdownTooltip.tsx

```diff
@@ -1,111 +1,92 @@
 import { Trans } from '@lingui/macro'
+import { Currency } from '@uniswap/sdk-core'
 import { AutoColumn } from 'components/Column'
 import UniswapXRouterLabel, { UniswapXGradient } from 'components/RouterLabel/UniswapXRouterLabel'
 import Row from 'components/Row'
+import { nativeOnChain } from 'constants/tokens'
 import { ReactNode } from 'react'
 import { InterfaceTrade } from 'state/routing/types'
-import { isClassicTrade, isUniswapXTrade } from 'state/routing/utils'
+import { isUniswapXTrade } from 'state/routing/utils'
 import styled from 'styled-components'
 import { Divider, ExternalLink, ThemedText } from 'theme/components'
 import { NumberType, useFormatter } from 'utils/formatNumbers'
 
-const Container = styled(AutoColumn)`
-  padding: 4px;
+const Container = styled(AutoColumn)<{ pad: boolean }>`
+  padding: ${({ pad }) => (pad ? '4px' : '0px')};
 `
 
-const InlineLink = styled(ThemedText.BodySmall)`
-  color: ${({ theme }) => theme.accent1};
-  display: inline;
-  cursor: pointer;
-  &:hover {
-    opacity: ${({ theme }) => theme.opacity.hover};
-  }
-`
-
-const InlineUniswapXGradient = styled(UniswapXGradient)`
-  display: inline;
-`
+type GasCostItemProps = { title: ReactNode; itemValue?: React.ReactNode; amount?: number }
 
-const GasCostItem = ({
-  title,
-  amount,
-  itemValue,
-}: {
-  title: ReactNode
-  itemValue?: React.ReactNode
-  amount?: number
-}) => {
+const GasCostItem = ({ title, amount, itemValue }: GasCostItemProps) => {
   const { formatNumber } = useFormatter()
 
+  if (!amount && !itemValue) return null
+
+  const value = itemValue ?? formatNumber({ input: amount, type: NumberType.FiatGasPrice })
   return (
     <Row justify="space-between">
       <ThemedText.SubHeaderSmall>{title}</ThemedText.SubHeaderSmall>
-      <ThemedText.SubHeaderSmall color="neutral1">
-        {itemValue ??
-          formatNumber({
-            input: amount,
-            type: NumberType.FiatGasPrice,
-          })}
-      </ThemedText.SubHeaderSmall>
+      <ThemedText.SubHeaderSmall color="neutral1">{value}</ThemedText.SubHeaderSmall>
     </Row>
   )
 }
 
-export function GasBreakdownTooltip({
-  trade,
-  hideFees = false,
-  hideUniswapXDescription = false,
-}: {
-  trade: InterfaceTrade
-  hideFees?: boolean
```

Comment:
the hideFees logic was used to allow this component to be either a `GasBreakdownTooltip` or a UniswapX description tooltip, which felt unecessarily-coupled so now `<UniswapXDescription /> ` can be used rather than a `<GasBreakdownTooltip hideFees="true" />`